test/generated_cases_test.exs

```diff
@@ -37,6 +37,10 @@ defmodule PublicSuffix.TestCaseGenerator do
       input: input,
       registrable_domain_output: registrable_domain_output,
       public_suffix_output: public_suffix_output(registrable_domain_output, input),
+      # prevailing_rule/2 should always return the same as public_suffix/2 except
+      # when the domain doesn't match an explicit rule, so the following should be fine,
```

Comment:
That sounds like a good idea. I'll have a look at it in the coming days and put together a new pull request (and also fix the other things you mentioned - thanks for the feedback!)